src/utils/cosmosdb_model_versioned.ts

```diff
@@ -166,6 +173,95 @@ export abstract class CosmosdbModelVersioned<
     );
   }
 
+  public updateTTLForAllVersions(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>,
+    ttl: NonNegativeNumber
+    // eslint-disable-next-line
+  ): TaskEither<CosmosErrors, TaskEither<never, readonly any[]>> {
+    return pipe(
+      this.findAllVersionsByPartitionKey(searchKey),
+      TE.map(RA.rights),
+      TE.map(
+        RA.map(doc => ({
+          id: doc.id,
+          operationType: "Patch" as const,
+          resourceBody: {
+            operations: [
+              {
+                /**
+                    add will replace the path with the passed value if the path alreadye exists,
+                    otherwise it will add the path
+                 */
+                op: PatchOperationType.add,
+                path: `/ttl`,
+                value: ttl
+              }
+            ]
+          }
+        }))
+      ),
+      TE.map(RA.chunksOf(100)),
+      TE.map(
+        RA.map(x =>
+          pipe(x, RA.toArray, operations =>
+            super.batch(
+              operations,
+              `${
+                searchKey.length === 1 ? searchKey[0] : searchKey[1]
+              }` as NonEmptyString
+            )
+          )
+        )
+      ),
+      TE.chainW(TE.sequenceSeqArray),
+      TE.chainW(responses =>
+        pipe(
+          responses,
+          RA.filter(response => response.code !== 200),
+          TE.fromPredicate(
+            errors => errors.length === 0,
+            _ =>
+              CosmosErrorResponse({
+                message: `Error updating ttl for ${searchKey}`,
+                name: `Error updating ttl`
+              })
+          ),
+          TE.map(() => TE.right(responses))
+        )
+      )
+    );
+  }
+
+  /**
+   * Given a searchKey returns all the version of a document
+   */
+
+  public findAllVersionsByPartitionKey(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>
+  ): TE.TaskEither<CosmosErrors, ReadonlyArray<t.Validation<TR>>> {
+    const partitionKey = searchKey.length === 1 ? searchKey[0] : searchKey[1];
+    return pipe(
+      TE.tryCatch(
+        () =>
+          asyncIterableToArray(
+            this.getQueryIterator({
+              parameters: [
+                {
+                  name: "@partitionKey",
+                  value: partitionKey as JSONValue
+                }
+              ],
+              query: `SELECT * FROM m WHERE m.${
```

Comment:
If you don't need to get all the fields consider listing them as an alternative to select star, since this will require more network and memory (resulting in a slower query).

Reply:
I agree with @gquadrati, this is a good optimization but I would move this to another pr, at the moment is not possible to filter fields because we pass  the query to `getQueryIterator` which calls the `TR.decode` on the results, so we can only pass the result of a `SELECT *`. I think it's cleaner to merge this pr as is and apply this change into another one.